### DISCUSSION TITLE
A (not-so-)spammy way to spread awareness

### DISCUSSION THREAD

**Discussion by nikolay** [aquaproj/aqua#2312] [Deprecated (Proposal)]
### Overview

Aqua is a great tool. It's much faster than `asdf` (even with the Rust replacement [rtx](https://github.com/jdx/rtx)), and safer, but it's not popular. As the repositories, which are supported, are in the standard registry, maybe an automated script to create an issue with the project, which suggests adding an installation with aqua (maybe even posting a small piece of suggested markdown to be included in the installation section.

### Why is the proposal needed?

Bring more awareness and support for the fantastic tool.

**Comment by nikolay** [comment]
Here's an example: https://github.com/Boeing/config-file-validator/issues/57

**Comment by suzuki-shunsuke** [comment]
Thank you for your suggestion.
Interesting.
We can create issues to repositories that aqua-registry is supporting by script.
We can add the installation guide to the issue description so that maintainers can copy and paste it to README easily.
I think it's not so difficult.

But I'm afraid that it may be spammy.
The script just creates a single issue, so maybe it's not so spammy. 🤔

**Reply by suzuki-shunsuke** [reply]
I wrote a template of issues.

# Add the installation guide with aqua to the document

_This issue was created automatically by script. For the detail, please see [here](https://github.com/orgs/aquaproj/discussions/2312)._

Hi, thank you for your great project!

We suggest to add the installation guide with aqua to the document.
[aqua](https://aquaproj.github.io/) is a declarative CLI Version Manager.
aqua has supported this tool, so you can install this tool with aqua.

```sh
aqua init # Create aqua.yaml
aqua g -i <package name> # Add this tool to aqua.yaml
aqua i # Install tools
```

For the detail of aqua, please see [the official document](https://aquaproj.github.io/).

We expect you can just copy and paste the following guide to your document.

---

## Install with aqua

You can install this tool with [aqua](https://aquaproj.github.io/).

```sh
aqua g -i <package name> # Add this tool to aqua.yaml
```

---

Thank you.

**Reply by nikolay** [reply]
Maybe somehow we can split repositories among supporters and run a script that posts from our own accounts.

**Comment by suzuki-shunsuke** [comment]
## Create repository lists

The list isn't perfect, but almost good.

```sh
git ls-files | grep -E "^pkgs/[^.]+/registry\.yaml$" | sed -E "s|^pkgs/([^./]+/[^./]+)[^.]*/registry\.yaml$|\1|" | sort -u
```

```console
$ git ls-files | grep -E "^pkgs/[^.]+/registry\.yaml$" | sed -E "s|^pkgs/([^./]+/[^./]+)[^.]*/registry\.yaml$|\1|" | sort -u | wc -l
    1183
```

## Create issues

- create_issues.sh
- repos.txt
- created_repos.txt
- body.md

```sh
#!/usr/bin/env bash

set -euo pipefail

echo "[INFO] Creating issues" >&2
while read -r repo; do
	if echo "$repo" | grep -f created_repos.txt > /dev/null; then
		echo "[INFO] Skipped creating an issue because it was already created to this repository  repo=$repo" >&2
		continue
	fi
	echo "[INFO] Creating an issue  repo=$repo" >&2
	sed "s|<package>|$repo|" body.md |
		gh -R "$repo" issue create \
			-F - \
			-t "Add the installation guide with aqua to the document"
	echo "$repo" >> created_repos.txt
done < repos.txt
echo "[INFO] Completed creating issues" >&2
```

**Reply by suzuki-shunsuke** [reply]
List only packages whose names are `<repo owner>/<repo name>`

```sh
git ls-files | grep -E "^pkgs/[^./]+/[^./]+/registry\.yaml$" | sed -E "s|^pkgs/(.*)/registry\.yaml$|\1|" | sort -u
```

**Reply by suzuki-shunsuke** [reply]
I created a repository. https://github.com/aquaproj/create-installation-guide-issues
And I created some issues with my OSS for testing.
It looks good.

```console
$ ./create_issues.sh 
[INFO] Creating issues
[INFO] Creating an issue  repo=suzuki-shunsuke/ci-info url=https://github.com/suzuki-shunsuke/ci-info

Creating issue in suzuki-shunsuke/ci-info

https://github.com/suzuki-shunsuke/ci-info/issues/401
[INFO] Skipped creating an issue because it was already created to this repository  repo=suzuki-shunsuke/circleci-config-merge url=https://github.com/suzuki-shunsuke/circleci-config-merge
[INFO] Creating an issue  repo=suzuki-shunsuke/cmdx url=https://github.com/suzuki-shunsuke/cmdx

Creating issue in suzuki-shunsuke/cmdx

https://github.com/suzuki-shunsuke/cmdx/issues/538
[INFO] Completed creating issues
```

- https://github.com/suzuki-shunsuke/ci-info/issues/401
- https://github.com/suzuki-shunsuke/cmdx/issues/538

**Reply by suzuki-shunsuke** [reply]
Before creating issues to all repositories, we can create only 10 ~ 20 issues and check the reaction first.

**Reply by suzuki-shunsuke** [reply]
# List of Issues

https://github.com/search?q=is%3Aissue%20author%3Asuzuki-shunsuke%20%22Add%20the%20installation%20guide%20with%20aqua%20to%20the%20document%22%20in%3Atitle&type=issues

**Reply by suzuki-shunsuke** [reply]
I created 13 issues first.

- Songmu/ecschedule
- fujiwara/lambroll
- goccy/bigquery-emulator
- hirosassa/ksnotify
- hirosassa/tfcmt-gitlab
- int128/kauthproxy
- k1LoW/runn
- kayac/ecspresso
- ryoppippi/zigchat
- siderolabs/conform
- siderolabs/talos
- takumin/gjson
- takumin/gyaml

**Comment by suzuki-shunsuke** [comment]
# Create GitHub Issues by script

Hi, thank you for your great project!
We created GitHub Issues to your project by [script](https://github.com/aquaproj/create-installation-guide-issues/blob/main/create_issues.sh).
We are sorry about that if you feel issues are noisy, spammy, or uncomfortable 🙇‍♂️

aqua is a declarative CLI Version Manager written in Go.
You can install command line tools and manage their versions with declarative YAML configuration files `aqua.yaml`.
aqua is similar to [asdf](https://asdf-vm.com/), but has various additional features such as [Renovate Support](https://aquaproj.github.io/docs/guides/renovate), [Checksum Verification](https://aquaproj.github.io/docs/reference/security/checksum), and [Lazy Install](https://aquaproj.github.io/docs/reference/lazy-install).
aqua supports over 1,200 packages! For the detail, please see [the official document](https://aquaproj.github.io/).

If you have any feedbacks on the issue creation, please comment!
If you want to improve the script or issue template, please send pull requests to https://github.com/aquaproj/create-installation-guide-issues .

- [script](https://github.com/aquaproj/create-installation-guide-issues/blob/main/create_issues.sh)
- [template](https://github.com/aquaproj/create-installation-guide-issues/blob/main/body.md)
